src/Sylius/Bundle/CoreBundle/Order/Checker/OrderPromotionsIntegrityChecker.php

```diff
@@ -11,31 +11,40 @@
 
 declare(strict_types=1);
 
-namespace Sylius\Bundle\ApiBundle\Checker;
+namespace Sylius\Bundle\CoreBundle\Order\Checker;
 
 use Doctrine\Common\Collections\ArrayCollection;
 use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\PromotionInterface;
 use Sylius\Component\Order\Processor\OrderProcessorInterface;
 
-/** @experimental */
-final class OrderIntegrityChecker implements OrderIntegrityCheckerInterface
+final class OrderPromotionsIntegrityChecker implements OrderPromotionsIntegrityCheckerInterface
 {
+    private ?PromotionInterface $removedPromotion = null;
+
     public function __construct(private OrderProcessorInterface $orderProcessor)
     {
     }
 
-    public function check(OrderInterface $order): void
+    public function check(OrderInterface $order): bool
     {
         $previousPromotions = new ArrayCollection($order->getPromotions()->toArray());
 
         $this->orderProcessor->process($order);
 
         foreach ($previousPromotions as $previousPromotion) {
             if (!$order->getPromotions()->contains($previousPromotion)) {
-                throw new \RuntimeException(
-                    sprintf('Order is no longer eligible for this promotion %s.', $previousPromotion->getName())
-                );
+                $this->removedPromotion = $previousPromotion;
```

Comment:
You shouldn't save this value on the checker, if we need a promotion name, we should probably return this name when there is not eligible promotion, and null when everything is ok 🤔

Reply:
Please check if I understood you correctly 🤔